File: 114-FlattenBinaryTreetoLinkedList/114-FlattenBinaryTreetoLinkedList.py
# Last updated: 2/14/2026, 3:47:47 PM
1# Definition for a binary tree node.
2class TreeNode:
3    def __init__(self, val=0, left=None, right=None):
4        self.val = val
5        self.left = left
6        self.right = right
7
8class Solution:
9    def flatten(self, root: TreeNode) -> None:
10        if not root:
11            return
12        
13        if root.left:
14            temp = root.right
15            root.right = root.left
16            root.left = None
17            
18            iterator = root.right
19            while iterator.right:
20                iterator = iterator.right
21            
22            iterator.right = temp
23        
24        self.flatten(root.right)
25

File: 114-FlattenBinaryTreetoLinkedList/114-FlattenBinaryTreetoLinkedList.cpp
// Last updated: 2/13/2026, 10:25:25 PM
1/**
2 * Definition for a binary tree node.
3 * struct TreeNode {
4 *     int val;
5 *     TreeNode *left;
6 *     TreeNode *right;
7 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
8 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
9 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
10 * };
11 */
12class Solution {
13public:
14    void flatten(TreeNode* root) {
15        if(!root){
16            return;
17        }
18        if (root->left){
19            TreeNode *temp = root-> right;
20            root->right = root->left;
21            root->left = nullptr;
22            TreeNode* iterator = root->right;
23            while(iterator->right){
24                iterator = iterator-> right;
25            }
26            iterator->right = temp;
27            
28        }
29        flatten(root->right);
30        
31    }
32};